sims.mk

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/make -f
+
+########################################
+### Simulations
+
+BINDIR ?= $(GOPATH)/bin
+SIMAPP = ./tests/simulation
+
+
+# Run sim is a cosmos tool which helps us to run multiple simulations in parallel.
+runsim: $(BINDIR)/runsim
+$(BINDIR)/runsim:
+	@echo 'Installing runsim...'
+	@TEMP_DIR=$$(mktemp -d) && \
+	cd $$TEMP_DIR && \
+	go install github.com/cosmos/tools/cmd/runsim@v1.0.0 && \
+	rm -rf $$TEMP_DIR || (echo 'Failed to install runsim' && exit 1)
+	@echo 'runsim installed successfully'
+
+
+define run-sim-test
+	@echo "Running $(1)..."
+	@go test -mod=readonly $(SIMAPP) -run $(2) -Enabled=true \
+		-NumBlocks=$(3) -BlockSize=$(4) -Commit=true -Period=0 -v -timeout $(5)
+endef
+
+test-sim-nondeterminism:
+	$(call run-sim-test,"non-determinism test",TestAppStateDeterminism,100,200,2h)
```

Comment:
Does it mean it run 200 batch? Why do we need that many?

Reply:
While testing, I noticed that sometimes the test only fails for larger batch sizes; this is not a finalized number and can be tweaked if needed.

Reply:
Will need to be decreased for CI integration